vim-canvas-demo-app-react/src/components/update-fields/inputField.tsx

```diff
@@ -32,17 +32,38 @@ export const InputField = ({
     }, 0);
   };
 
+  const numberInputValidator = useCallback((evt: React.KeyboardEvent<HTMLInputElement>) => {
+    if(inputType === 'number') {
+      if(isValidNumberChar(evt.key)) {
+        evt.preventDefault();
+      }
+    }
+    
+  }, [inputType])
+
+
+  const numberInputPasteValidator = useCallback((evt: React.ClipboardEvent<HTMLInputElement>) => {
+    if(inputType === 'number') {
+      const clipboardData = evt.clipboardData?.getData('text');
+      if(clipboardData && isValueAllNumbers(clipboardData)) {
+        evt.preventDefault();
+      }
+    }
+  }, [inputType])
+
   return (
     <div className="flex w-full justify-between">
       <div className="relative w-full">
         <Input
           type={inputType}
-          pattern={pattern}
           className="h-7 rounded-r-none"
           value={innerValue}
           onChange={(e) => setInnerValue(e.target.value)}
           disabled={!editMode}
           ref={inputRef}
+          onKeyDown={numberInputValidator}
```

Comment:
I can still add decimal numbers (do we want to allow it?)
![image](https://github.com/user-attachments/assets/0a21efbe-dbda-4df3-a6c6-bede0dee8858)

And also negative numbers by using the down arrow
![image](https://github.com/user-attachments/assets/e0996c55-6d9f-4230-aa50-1b36d6539183)